### ISSUE TITLE
"npm install" w. warnings & "npm run examples:serve" w. error

### ISSUE THREAD

**Issue by zhangzlili** [Shopify/js-buy-sdk#284]
After doing "git clone git@github.com:Shopify/js-buy-sdk.git" to my vps hosting site in the folder /home/posnowz/js-buy-sdk, I did "npm install" which output a few warnings. Ignoring the warnings I did "npm run examples:serve" which output many errors and failed. It said that these errors are not related to npm but not finding some shopify files.

So I include the output as the followings:

```bash
[posnowz@vps ~]$ cd js-buy-sdk
[posnowz@vps js-buy-sdk]$ ls
CHANGELOG.md        DEPLOYING.md  circle.yml  package.json   service.yml  testem.json
CODE_OF_CONDUCT.md  LICENSE.txt   docs        scripts        shipit.yml   tests
```
After doing "git clone git@github.com:Shopify/js-buy-sdk.git" to my vps hosting site in the folder /home/posnowz/js-buy-sdk, I did "npm install" which output a few warnings. Ignoring the warnings I did "npm run examples:serve" which output many errors and failed. It said that these errors are not related to npm but not finding some shopify files.

So I include the output as the followings:

```bash
[posnowz@vps ~]$ cd js-buy-sdk
[posnowz@vps js-buy-sdk]$ ls
CHANGELOG.md        DEPLOYING.md  circle.yml  package.json   service.yml  testem.json
CODE_OF_CONDUCT.md  LICENSE.txt   docs        scripts        shipit.yml   tests
CONTRIBUTING.md     README.md     examples    secrets.ejson  src          yuidoc.json
[posnowz@vps js-buy-sdk]$ npm install
Enter passphrase for key '/home/posnowz/.ssh/id_rsa':
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated socket.io-pure@1.3.12: socket.io-pure is no longer maintained as the original socket.io removed all native dependencies. Please use the original.
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated socket.io-client-pure@1.3.12: socket.io-client-pure is no longer maintained as the original socket.io removed all native dependencies. Please use the original.
npm WARN deprecated engine.io-pure@1.5.9: engine.io-purce is no longer maintained as the original engine.io removed all native dependencies. Please use the original.
npm WARN deprecated ws-pure@0.8.0: ws-pure is no longer maintained as the original ws removed all native dependencies. Please use the original ws.
npm WARN deprecated engine.io-client-pure@1.5.9: engine-io-client-pure is no longer maintained as the original engine-io-client removed all native dependencies. Please use the original.
npm WARN prefer global node-gyp@3.4.0 should be installed with -g

> fibers@1.0.15 install /home/posnowz/js-buy-sdk/node_modules/fibers
> node build.js || nodejs build.js



[posnowz@vps js-buy-sdk]$ npm run examples:serve

> shopify-buy@0.5.4 examples:serve /home/posnowz/js-buy-sdk
> rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'


> shopify-buy@0.5.4 examples:build /home/posnowz/js-buy-sdk
> npm run examples:copy && npm run examples-css:build


> shopify-buy@0.5.4 examples:copy /home/posnowz/js-buy-sdk
> cp -r examples .dist-test/examples


> shopify-buy@0.5.4 examples-css:build /home/posnowz/js-buy-sdk
> concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'

[1]
[1] > shopify-buy@0.5.4 examples-css-checkout:build /home/posnowz/js-buy-sdk
[1] > node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1]
[0]
[0] > shopify-buy@0.5.4 examples-css-cart:build /home/posnowz/js-buy-sdk
[0] > node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css
[0]
[1] fs.js:808
[1]   return binding.readdir(pathModule._makeLong(path));
[1]                  ^
[1]
[1] Error: ENOENT: no such file or directory, scandir '/home/posnowz/js-buy-sdk/node_modules/node-sass/vendor'
[1]     at Error (native)
[1]     at Object.fs.readdirSync (fs.js:808:18)
[1]     at Object.getInstalledBinaries (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/extensions.js:119:13)
[1]     at foundBinariesList (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:20:15)
[1]     at foundBinaries (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:15:5)
[1]     at Object.module.exports.missingBinary (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:45:5)
[1]     at Object.<anonymous> (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/index.js:15:28)
[1]     at Module._compile (module.js:409:26)
[1]     at Object.Module._extensions..js (module.js:416:10)
[1]     at Module.load (module.js:343:32)
[1]
[1] npm
[1]  ERR! Linux 2.6.32-042stab092.2
[1] npm
[1]  ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples-css-checkout:build"
[1] npm ERR! node
[1]  v4.6.2
[1] npm ERR! npm  v4.0.3
[1] npm ERR! code
[1]  ELIFECYCLE
[1] npm ERR! shopify-buy@0.5.4 examples-css-checkout:build: `node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css`
[1] npm ERR! Exit status 1
[1] npm ERR!
[1] npm ERR! Failed at the shopify-buy@0.5.4 examples-css-checkout:build script 'node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css'.
[1] npm
[1] ERR! Make sure you have the latest version of node.js and npm installed.
[1] npm ERR! If you do, this is most likely a problem with the shopify-buy package,
[1] npm ERR! not with npm itself.
[1] npm ERR! Tell the author that this fails on your system:
[1] npm ERR!     node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1] npm ERR! You can get information on how to open an issue for this project with:
[1] npm
[1]  ERR!     npm bugs shopify-buy
[1] npm ERR! Or if that isn't available, you can get their info via:
[1] npm ERR!     npm owner ls shopify-buy
[1] npm ERR! There is likely additional logging output above.
[1]
[1] npm ERR! Please include the following file with any support request:
[1] npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log
[1] npm run examples-css-checkout:build exited with code 1
[0] fs.js:808
[0]   return binding.readdir(pathModule._makeLong(path));
[0]                  ^
[0]
[0] Error: ENOENT: no such file or directory, scandir '/home/posnowz/js-buy-sdk/node_modules/node-sass/vendor'
[0]     at Error (native)
[0]     at Object.fs.readdirSync (fs.js:808:18)
[0]     at Object.getInstalledBinaries (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/extensions.js:119:13)
[0]     at foundBinariesList (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:20:15)
[0]     at foundBinaries (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:15:5)
[0]     at Object.module.exports.missingBinary (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/errors.js:45:5)
[0]     at Object.<anonymous> (/home/posnowz/js-buy-sdk/node_modules/node-sass/lib/index.js:15:28)
[0]     at Module._compile (module.js:409:26)
[0]     at Object.Module._extensions..js (module.js:416:10)
[0]     at Module.load (module.js:343:32)
[0]
[0] npm
[0]  ERR! Linux 2.6.32-042stab092.2
[0] npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples-css-cart:build"
[0] npm ERR! node v4.6.2
[0] npm ERR!
[0]  npm  v4.0.3
[0] npm ERR! code ELIFECYCLE
[0] npm
[0] ERR! shopify-buy@0.5.4 examples-css-cart:build: `node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css`
[0] npm ERR! Exit status 1
[0] npm ERR!
[0]
[0] npm ERR! Failed at the shopify-buy@0.5.4 examples-css-cart:build script 'node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css'.
[0] npm ERR! Make sure you have the latest version of node.js and npm installed.
[0] npm ERR! If you do, this is most likely a problem with the shopify-buy package,
[0] npm ERR! not with npm itself.
[0] npm ERR! Tell the author that this fails on your system:
[0] npm
[0] ERR!     node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css
[0] npm ERR! You can get information on how to open an issue for this project with:
[0] npm ERR!
[0]      npm bugs shopify-buy
[0] npm ERR! Or if that isn't available, you can get their info via:
[0] npm ERR!     npm owner ls shopify-buy
[0] npm
[0]  ERR! There is likely additional logging output above.
[0]
[0] npm ERR! Please include the following file with any support request:
[0] npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log
[0] npm run examples-css-cart:build exited with code 1

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples-css:build"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples-css:build: `concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples-css:build script 'concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs shopify-buy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls shopify-buy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples:build"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples:build: `npm run examples:copy && npm run examples-css:build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples:build script 'npm run examples:copy && npm run examples-css:build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run examples:copy && npm run examples-css:build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs shopify-buy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls shopify-buy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples:serve"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples:serve: `rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples:serve script 'rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'
npm ERR! You can get information on how to open an issue for this project with:
CONTRIBUTING.md     README.md     examples    secrets.ejson  src          yuidoc.json
[posnowz@vps js-buy-sdk]$ npm install
Enter passphrase for key '/home/posnowz/.ssh/id_rsa':
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated socket.io-pure@1.3.12: socket.io-pure is no longer maintained as the original socket.io removed all native dependencies. Please use the original.
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated socket.io-client-pure@1.3.12: socket.io-client-pure is no longer maintained as the original socket.io removed all native dependencies. Please use the original.
npm WARN deprecated engine.io-pure@1.5.9: engine.io-purce is no longer maintained as the original engine.io removed all native dependencies. Please use the original.
npm WARN deprecated ws-pure@0.8.0: ws-pure is no longer maintained as the original ws removed all native dependencies. Please use the original ws.
npm WARN deprecated engine.io-client-pure@1.5.9: engine-io-client-pure is no longer maintained as the original engine-io-client removed all native dependencies. Please use the original.
npm WARN prefer global node-gyp@3.4.0 should be installed with -g

> fibers@1.0.15 install /home/posnowz/js-buy-sdk/node_modules/fibers
> node build.js || nodejs build.js

```

But is seems that there is no npm-npm-debug.log found.

Please help me out with this.

Thank you so much!

**Comment by minasmart** [comment]
It looks like there are some issues with node-sass installing properly. What is the output if you run `npm install node-sass` from the project root? If that completes successfully, try serving examples again.

**Comment by zhangzlili** [comment]
I typed the following:

[posnowz@vps js-buy-sdk]$ npm install node-sass

> node-sass@3.8.0 install /home/posnowz/js-buy-sdk/node_modules/node-sass
> node scripts/install.js

Binary downloaded and installed at /home/posnowz/js-buy-sdk/node_modules/node-sass/vendor/linux-x64-46/binding.node

> node-sass@3.8.0 postinstall /home/posnowz/js-buy-sdk/node_modules/node-sass
> node scripts/build.js

"/home/posnowz/js-buy-sdk/node_modules/node-sass/vendor/linux-x64-46/binding.node" exists.
 testing binary.
Binary is fine; exiting.
shopify-buy@0.5.4 /home/posnowz/js-buy-sdk
`-- node-sass@3.8.0

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})


Then I typed the following:

[posnowz@vps js-buy-sdk]$ npm run examples:serve

> shopify-buy@0.5.4 examples:serve /home/posnowz/js-buy-sdk
> rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'


> shopify-buy@0.5.4 examples:build /home/posnowz/js-buy-sdk
> npm run examples:copy && npm run examples-css:build


> shopify-buy@0.5.4 examples:copy /home/posnowz/js-buy-sdk
> cp -r examples .dist-test/examples


> shopify-buy@0.5.4 examples-css:build /home/posnowz/js-buy-sdk
> concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'

[0]
[0] > shopify-buy@0.5.4 examples-css-cart:build /home/posnowz/js-buy-sdk
[0] > node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css
[0]
[1]
[1] > shopify-buy@0.5.4 examples-css-checkout:build /home/posnowz/js-buy-sdk
[1] > node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1]
[1] sh: line 1:  1937 Aborted                 node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1]
[1] npm
[1]  ERR! Linux 2.6.32-042stab092.2
[1] npm ERR!
[1] argv
[1]  "/usr/bin/node" "/usr/bin/npm" "run" "examples-css-checkout:build"
[1] npm ERR! node v4.6.2
[1] npm
[1]  ERR! npm  v4.0.3
[1] npm ERR! code
[1]  ELIFECYCLE
[1] npm ERR! shopify-buy@0.5.4 examples-css-checkout:build: `node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css`
[1] npm ERR!
[1]  Exit status 134
[1] npm ERR!
[1] npm ERR!
[1]  Failed at the shopify-buy@0.5.4 examples-css-checkout:build script 'node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css'.
[1] npm ERR! Make sure you have the latest version of node.js and npm installed.
[1] npm ERR! If you do, this is most likely a problem with the shopify-buy package,
[1] npm
[1] ERR! not with npm itself.
[1] npm ERR! Tell the author that this fails on your system:
[1] npm
[1]  ERR!     node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1] npm ERR! You can get information on how to open an issue for this project with:
[1] npm ERR!     npm bugs shopify-buy
[1] npm ERR! Or if that isn't available, you can get their info via:
[1] npm ERR!     npm owner ls shopify-buy
[1] npm ERR! There is likely additional logging output above.
[1]
[1] npm ERR! Please include the following file with any support request:
[1] npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log
[0] npm run examples-css-cart:build exited with code 0
[1] npm run examples-css-checkout:build exited with code 1

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples-css:build"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples-css:build: `concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples-css:build script 'concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs shopify-buy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls shopify-buy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples:build"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples:build: `npm run examples:copy && npm run examples-css:build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples:build script 'npm run examples:copy && npm run examples-css:build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run examples:copy && npm run examples-css:build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs shopify-buy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls shopify-buy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log

npm ERR! Linux 2.6.32-042stab092.2
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "examples:serve"
npm ERR! node v4.6.2
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-buy@0.5.4 examples:serve: `rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shopify-buy@0.5.4 examples:serve script 'rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the shopify-buy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs shopify-buy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls shopify-buy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log
[posnowz@vps js-buy-sdk]$ ls n*.*
npm-debug.log
[posnowz@vps js-buy-sdk]$


[npm-debug.txt](https://github.com/Shopify/js-buy-sdk/files/690212/npm-debug.txt)

**Comment by minasmart** [comment]
OK, so two things are standing out to me. Our examples are supported under the current version of node (the library itself is compatible with current and LTS). Could you try running under node current (`v6.0.0`)? However, it looks like there's an underlying problem with node-sass. Do you know if node-sass supports the OS you're using? Also what OS and version are you using?

**Comment by zhangzlili** [comment]
I am upgrading to node current v6.0.0. And my hosting site OS is CentOS Linux 6.8

**Comment by minasmart** [comment]
The library itself should work just fine there as a dependency, however the examples may not work on that OS. They're only for development use, which means desktop OSes (OS X, ubuntu linux, windows)

**Comment by zhangzlili** [comment]
I have upgraded node to v6.9.4 and rerun the followings:

#npm install

which produced a few warnings. Ignoring the warnings I did:

#npm install node-sass

which produced a few warnings too. Ignoring the warnings I did:

#npm run examples:serve

which produced a list of errors, saying that there are some problem not relating to npm but to js-shopify-buy.

The detail output is the following:

[root@vps posnowz]# cd js-buy-sdk
[root@vps js-buy-sdk]# ls
CHANGELOG.md        DEPLOYING.md  circle.yml  examples       package.json   service.yml  testem.json
CODE_OF_CONDUCT.md  LICENSE.txt   dist        node_modules   scripts        shipit.yml   tests
CONTRIBUTING.md     README.md     docs        npm-debug.log  secrets.ejson  src          yuidoc.json

[root@vps js-buy-sdk]# npm install

npm WARN prepublish-on-install As of npm@5, `prepublish` scripts will run only for `npm publish`.
npm WARN prepublish-on-install (In npm@4 and previous versions, it also runs for `npm install`.)
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
npm WARN lifecycle shopify-buy@0.5.4~prepublish: cannot run in wd %s %s (wd=%s) shopify-buy@0.5.4 npm run build /home/posnowz/js-buy-sdk
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

[root@vps js-buy-sdk]# npm install node-sass

> node-sass@3.8.0 install /home/posnowz/js-buy-sdk/node_modules/node-sass
> node scripts/install.js

Binary downloaded and installed at /home/posnowz/js-buy-sdk/node_modules/node-sass/vendor/linux-x64-48/binding.node

> node-sass@3.8.0 postinstall /home/posnowz/js-buy-sdk/node_modules/node-sass
> node scripts/build.js

"/home/posnowz/js-buy-sdk/node_modules/node-sass/vendor/linux-x64-48/binding.node" exists.
 testing binary.
Binary is fine; exiting.
shopify-buy@0.5.4 /home/posnowz/js-buy-sdk
`-- node-sass@3.8.0
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

[root@vps js-buy-sdk]# npm run examples:serve

> shopify-buy@0.5.4 examples:serve /home/posnowz/js-buy-sdk
> rimraf .dist-test/; mkdir .dist-test/; rimraf dist/; mkdir dist/; npm run examples:build && concurrently 'npm run umd-shopify-buy:watch' 'http-server -p 4200 .dist-test/'
> shopify-buy@0.5.4 examples:build /home/posnowz/js-buy-sdk
> npm run examples:copy && npm run examples-css:build


> shopify-buy@0.5.4 examples:copy /home/posnowz/js-buy-sdk
> cp -r examples .dist-test/examples


> shopify-buy@0.5.4 examples-css:build /home/posnowz/js-buy-sdk
> concurrently 'npm run examples-css-cart:build' 'npm run examples-css-checkout:build'

[0]
[0] > shopify-buy@0.5.4 examples-css-cart:build /home/posnowz/js-buy-sdk
[0] > node-sass examples/cart/styles/styles.scss > .dist-test/examples/cart/index.css
[0]
[1]
[1] > shopify-buy@0.5.4 examples-css-checkout:build /home/posnowz/js-buy-sdk
[1] > node-sass examples/checkout/styles/styles.scss > .dist-test/examples/checkout/index.css
[1]
[1] npm run examples-css-checkout:build exited with code 0
[0] npm run examples-css-cart:build exited with code 0
[1] Starting up http-server, serving .dist-test/
[1] Available on:
[1]   http://127.0.0.1:4200
[1]   http://127.0.0.1:4200
[1]   http://65.60.54.133:4200
[1] Hit CTRL-C to stop the server
[0]
[0] > shopify-buy@0.5.4 umd-shopify-buy:watch /home/posnowz/js-buy-sdk
[0] > nodemon --exec 'npm run umd-shopify-buy-all:build && cp dist/*.js .dist-test/' -w src/
[0]
[0] [nodemon] 1.10.0
[0] [nodemon] to restart at any time, enter `rs`
[0] [nodemon] watching: /home/posnowz/js-buy-sdk/src/**/*
[0] [nodemon] starting `npm run umd-shopify-buy-all:build && cp dist/*.js .dist-test/`
[0]
[0] > shopify-buy@0.5.4 umd-shopify-buy-all:build /home/posnowz/js-buy-sdk
[0] > concurrently 'npm run umd-shopify-buy:build' 'npm run umd-shopify-buy-polyfilled:build'
[0]
[0] [0]
[0] [0] > shopify-buy@0.5.4 umd-shopify-buy:build /home/posnowz/js-buy-sdk
[0] [0] > rollup -f cjs src/shopify.js | node scripts/runBabel umd $(get-project-version --template v{{version}}-{{commit}}) > dist/shopify-buy.umd.js
[0] [0]
[0] [0]
[0] [1]
[0] [1] > shopify-buy@0.5.4 umd-shopify-buy-polyfilled:build /home/posnowz/js-buy-sdk
[0] [1] > rollup -f cjs --config .rollup.config.js src/shopify-polyfilled.js | node scripts/runBabel umd $(get-project-version --template v{{version}}-{{commit}}) > dist/shopify-buy.umd.polyfilled.js
[0] [1]
[0] [0]
[0] [1]
[0] [0] npm run umd-shopify-buy:build exited with code 1
[0] [1]
[0] [1] npm run umd-shopify-buy-polyfilled:build exited with code 1
[0]
[0] npm
[0]  ERR! Linux 2.6.32-042stab092.2
[0] npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "umd-shopify-buy-all:build"
[0] npm ERR! node v6.9.4
[0] npm ERR! npm  v4.0.5
[0] npm ERR! code ELIFECYCLE
[0] npm ERR! shopify-buy@0.5.4 umd-shopify-buy-all:build: `concurrently 'npm run umd-shopify-buy:build' 'npm run umd-shopify-buy-polyfilled:build'`
[0] npm ERR! Exit status 1
[0] npm ERR!
[0] npm ERR! Failed at the shopify-buy@0.5.4 umd-shopify-buy-all:build script 'concurrently 'npm run umd-shopify-buy:build' 'npm run umd-shopify-buy-polyfilled:build''.
[0] npm ERR! Make sure you have the latest version of node.js and npm installed.
[0] npm ERR! If you do, this is most likely a problem with the shopify-buy package,
[0] npm ERR! not with npm itself.
[0] npm ERR! Tell the author that this fails on your system:
[0] npm ERR!     concurrently 'npm run umd-shopify-buy:build' 'npm run umd-shopify-buy-polyfilled:build'
[0] npm ERR! You can get information on how to open an issue for this project with:
[0] npm ERR!     npm bugs shopify-buy
[0] npm ERR! Or if that isn't available, you can get their info via:
[0] npm ERR!     npm owner ls shopify-buy
[0] npm ERR! There is likely additional logging output above.
[0]
[0] npm ERR! Please include the following file with any support request:
[0] npm ERR!     /home/posnowz/js-buy-sdk/npm-debug.log
[0] [nodemon] app crashed - waiting for file changes before starting...

[npm-debug.txt](https://github.com/Shopify/js-buy-sdk/files/700029/npm-debug.txt)

Please help me out with this.
Thank you so much!

**Comment by zhangzlili** [comment]
Can anyone help me with my issue? I am almost dead, going nowhere and completely lost.

Thank you so much!

**Comment by zhangzlili** [comment]
At least give me some direction.

**Comment by mikkoh** [comment]
@zhangzlili maybe I can help? What is you're trying to achieve with the `js-buy-sdk`? Maybe we can figure out a more direct route than what you're currently taking.

Are you wanting to incorporate the `js-buy-sdk` into an existing application? Does this project use `npm` for a module management?

or

Are you simply wanting to learn how to work with the `js-buy-sdk`?

or 

Some other use case?

**Comment by zhangzlili** [comment]
Yes! 

1st, I am trying to run js-buy-sdk's checkout function on my vps hosting site.
2nd, I am going to incorporate this function into my existing simple e-commerce website written in php & mysql to complete each transaction in real time.

My existing project does not use npm for a module management yet.

Thank you so much!

**Comment by mikkoh** [comment]
Ok. So since you're attempting to incorporate into an existing `php` site maybe the easiest way to do this is to include the `js-buy-sdk` in a new html `<script>` tag instead of consuming via`npm`.

To do this in your site include the following html script tag into the generated html from php:
`<script src="http://sdks.shopifycdn.com/js-buy-sdk/v0/latest/shopify-buy.umd.polyfilled.min.js"></script>`

After this to check everything is working you could create a second script tag that would look like this:
```html
<script>
const shopClient = ShopifyBuy.buildClient({
  apiKey: 'bf081e860bc9dc1ce0654fdfbc20892d',
  appId: 6,
  domain: 'embeds.myshopify.com'
});
 
shopClient.fetchAllProducts()
.then(function (products) {
  console.log(products);
})
.catch(function () {
  console.log('Request failed');
});
</script>
```

The above script will request all products from Shopify's testing shop. In Google Chrome's Developer tools you should see an `Array` of products output.

The next step after this would be to replace the `apiKey` and `domain` with your Shopify shop credentials in the above script. You should be able to find details on how to acquire an `apiKey` from this link:
https://help.shopify.com/api/sdks/js-buy-sdk/getting-started#app-id

Once you've modified `apiKey` and `domain` you'll be querying all products from your Shop and should be able to double check everything is working via Google Chrome's Developer Tools.

Beyond this you should be able to follow the documentation to create your application:
https://shopify.github.io/js-buy-sdk/

The above should also explain how checkouts are currently handled by the `js-buy-sdk`.

**Comment by zhangzlili** [comment]
To test the checkout function, I wrote index.html as the following:

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Buy SDK Example -- Cart 001</title>
   <script src="http://sdks.shopifycdn.com/js-buy-sdk/v0/latest/shopify-buy.umd.polyfilled.min.js"></script>

</head>
<body>

  <div class="cart">
    <div class="not-empty">
      <img class="variant-image">
      <h2><span class="line-item-count">0</span> items in cart</h2>
      <h2>Sub-total: <span class="sub-total">$0.00</span></h2>
      <button class="btn checkout">Checkout</button>
    </div>

    <div class="empty">
      <h2>Thanks for checking out!!</h2>
    </div>
  </div>

  <script>
  const shopClient = ShopifyBuy.buildClient({
    apiKey: 'bf081e860bc9dc1ce0654fdfbc20892d',
    appId: 6
    domain: 'embeds.myshopify.com'
  });

  shopifyClient.fetchAllProducts()
  .then(function (products){
    console.log(products);
  })
  .catch(function () {
    console.log("Request failed');
  });
  </script>

</body>
</html>


And run it from http://vps.posnowz.com/js-buy-sdk/examples/checkout/

Could you please help me to debug this?

Thanks!

**Comment by zhangzlili** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Buy SDK Example -- Cart 001</title>
  <link rel="stylesheet" href="index.css">
  <script src="http://sdks.shopifycdn.com/js-buy-sdk/v0/latest/shopify-buy.umd.polyfilled.min.js"></script>

</head>
<body>

  <div class="cart">
    <div class="not-empty">
      <img class="variant-image">
      <h2><span class="line-item-count">0</span> items in cart</h2>
      <h2>Sub-total: <span class="sub-total">$0.00</span></h2>
      <button class="btn checkout">Checkout</button>
    </div>

    <div class="empty">
      <h2>Thanks for checking out!!</h2>
    </div>
  </div>

  <script>
  const shopClient = ShopifyBuy.buildClient({
    apiKey: 'bf081e860bc9dc1ce0654fdfbc20892d',
    appId: 6
    domain: 'embeds.myshopify.com'
  });

  shopifyClient.fetchAllProducts()
  .then(function (products){
    console.log(products);
  })
  .catch(function () {
    console.log("Request failed');
  });
  </script>



</body>
</html>

**Comment by zhangzlili** [comment]
[index.txt](https://github.com/Shopify/js-buy-sdk/files/702660/index.txt)

**Comment by mikkoh** [comment]
You're JS code is currently syntactically invalid.

A couple of things. You create:
`const shopClient = ShopifyBuy.buildClient({`

however later on you attempt to use `shopClient` as `shopifyClient`:
`shopifyClient.fetchAllProducts()`

you should change `shopifyClient` to be `shopClient`. 

Another issue is that you're missing a comma after `appId: 6`.

```
    apiKey: 'bf081e860bc9dc1ce0654fdfbc20892d',
    appId: 6
    domain: 'embeds.myshopify.com'
```

Should be:
```
    apiKey: 'bf081e860bc9dc1ce0654fdfbc20892d',
    appId: 6,
    domain: 'embeds.myshopify.com'
```

Another portion of your code which is syntactically invalid is that you are mixing `"` and `'` when creating a string here:
```
console.log("Request failed');
```
This should be:
```
console.log("Request failed");
```

**Comment by zhangzlili** [comment]
I have debugged these in the index.html. Could you please give me instructions on how to test the checkout function?

As I went to the site  http://vps.posnowz.com/js-buy-sdk/examples/checkout/ and clicked the checkout button and nothing happened.

[index.txt](https://github.com/Shopify/js-buy-sdk/files/702767/index.txt)

**Comment by mikkoh** [comment]
In your above example you will need to update `domain` to be the domain for your Shopify shop. You will also need to setup `apiKey` with your shops generated api key.

Once the above is done and after reading over the documentation and following examples in the following urls:
http://shopify.github.io/js-buy-sdk/guide/
https://github.com/Shopify/js-buy-sdk/tree/master/examples/checkout

You'll have enough knowledge to create your application.

I will close this issue now as you should have enough information to be able to develop with the `js-buy-sdk`
